File: nuxeo-opensocial-features/src/main/js/nuxeo/nuxeo.js
/*
 * (C) Copyright 2006-2009 Nuxeo SA (http://nuxeo.com/) and contributors.
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the GNU Lesser General Public License
 * (LGPL) version 2.1 which accompanies this distribution, and is available at
 * http://www.gnu.org/licenses/lgpl.html
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * Contributors:
 *     Leroy Merlin (http://www.leroymerlin.fr/) - initial implementation
 */

var gadgets = gadgets || {};

gadgets.nuxeo = gadgets.nuxeo || {};

gadgets.nuxeo.URLBASE = top.nxContextPath + "/site/gadgetDocumentAPI/";
gadgets.nuxeo.HTMLACTION = "/html";
gadgets.nuxeo.FILEACTION =  "/file";


/**
 * Get the url for the gadget's Resource
 *
 * @scope gadgets.nuxeo
 */
gadgets.nuxeo.getResourceUrl = function(id) {
  return gadgets.nuxeo.URLBASE + id;
}


/**
* Get the url for the gadget's HTML action
*
* @scope gadgets.nuxeo
*/
gadgets.nuxeo.getHtmlActionUrl = function(id) {
  return this.getResourceUrl(id) + this.HTMLACTION;
}

/**
* Get the url for the gadget's Form action
*
* @scope gadgets.nuxeo
*/
gadgets.nuxeo.getFormActionUrl = function(id) {
  return this.getResourceUrl(id);
}

gadgets.nuxeo.getFileActionUrl = function(id) {
  return this.getResourceUrl(id) + this.FILEACTION;
}

/**
 * Get the url to get a file with a specific name
 */
gadgets.nuxeo.getSpecificFileUrl = function(id, specificName) {
  return this.getResourceUrl(id) + this.FILEACTION + "/" + specificName;
}

/**
 * Returns the current gadget's id
 */
gadgets.nuxeo.getGadgetId = function() {
  return window.name.split("-").slice(2).join("-");
};

gadgets.nuxeo.hasPermission = function(permissionName) {
  return this.isEditable();
};

gadgets.nuxeo.isEditable = function() {
  return gadgets.util.getUrlParameters().permission == "1";
};

gadgets.nuxeo.getNXIDPreference = function(name, id) {
  return {"NXID": id, "NXNAME":name};
};

gadgets.nuxeo.refreshGadget = function(){
  gadgets.rpc.call("", "refresh", null, "");
};

File: nuxeo-distribution/nuxeo-jsf-ui-gatling-tests/scripts/nuxeo.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

# (C) Copyright 2015 Nuxeo SA (http://nuxeo.com/) and contributors.
#
# All rights reserved. This program and the accompanying materials
# are made available under the terms of the GNU Lesser General Public License
# (LGPL) version 2.1 which accompanies this distribution, and is available at
# http://www.gnu.org/licenses/lgpl-2.1.html
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# Contributors:
#     Delbosc Benoit
#

import os
import json
from urllib import pathname2url
from utils import truncate

def getNuxeoPayload(docType, name, properties):
    """ Returns the Nuxeo REST API payload to create a document"""
    return '{ "entity-type": "document", "name":"%s", "type": "%s","properties": %s}' % (
        name, docType, json.dumps(properties, ensure_ascii=False))


def getNuxeoFolderPayload(name):
    """ Create a payload for a folder"""
    return getNuxeoPayload("Folder", name, {"dc:title": name})


class NuxeoWriter(object):
    """ Helper to write a document and its parents into Redis
    """

    def __init__(self, writer):
        self.writer = writer

    def addDocument(self, name, docType, parentPath, properties):
        """ Add a document and all its parents
        """
        assert not '/' in name, 'name with slashes is forbiden: %s' % name
        self.addParents(parentPath)
        payload = getNuxeoPayload(docType, name, properties)
        key = "/".join((parentPath, name))
        self.writer.sadd("doc", key)
        self.writer.hmset("data:" + key, {"parentPath": pathname2url(parentPath), "type": docType, "name": name,
                                          "payload": payload, "key": key, "url": pathname2url(key)})

    def addParents(self, path):
        """ Add all parents folder for this path
        """
        segments = path.split("/")
        for i in range(len(segments)):
            self.addFolder("/".join(segments[0:i + 1]))

    def addFolder(self, path):
        """ Add a folder if it does not exists already
        """
        level = len(path.split("/"))
        name = os.path.basename(path)
        if (level > 1):
            parentPath = os.path.dirname(path)
        else:
            parentPath = ""
        payload = getNuxeoFolderPayload(name)
        self.writer.zadd("folder", level, path)
        key = "data:" + path
        self.writer.hsetnx(key, "parentPath", pathname2url(parentPath))
        self.writer.hsetnx(key, "name", name)
        self.writer.hsetnx(key, "type", "Folder")
        self.writer.hsetnx(key, "payload", payload)
        self.writer.hsetnx(key, "key", key)
        self.writer.hsetnx(key, "url", pathname2url(key))

